gradle/java-setup.gradle

```diff
@@ -34,23 +34,23 @@ eclipseResourceFilters {
 apply plugin: 'com.github.spotbugs'
 spotbugs {
 	toolVersion = VER_SPOTBUGS
-	sourceSets = [
-		// don't check the test code
-		sourceSets.main
-	]
 	ignoreFailures = false 	// bug free or it doesn't ship!
 	reportsDir = file('build/spotbugs')
 	effort = 'max'			// min|default|max
 	reportLevel = 'medium'	// low|medium|high (low = sensitive to even minor mistakes)
 	omitVisitors = []		// bugs that we want to ignore
 }
-// HTML instead of XML
-tasks.withType(spotBugsTaskType()) {
+tasks.named('spotbugsTest') {
+	enabled = false
+}
+spotbugsMain {
 	reports {
-		xml.enabled = false
-		html.enabled = true
+		html {
+			enabled = true
+		}
 	}
 }
+
 dependencies {
 	compileOnly 'net.jcip:jcip-annotations:1.0'
 	compileOnly 'com.github.spotbugs:spotbugs-annotations:3.1.6'
```

Comment:
It occurs to me that this dependency `spotbugs-annotations` might be out-of-sync with SpotBugs itself. Have you considered importing and running [`gradle-versions-plugin`](https://plugins.gradle.org/plugin/com.github.ben-manes.versions) (or IntelliJ's [Package Search](https://plugins.jetbrains.com/plugin/12507-package-search) plugin)?

Whilst we're at it, it might be worth upgrading all of Spotless's other dependencies as well. Thoughts?

Reply:
Good idea!  I'll take a dep pass.  If anything gets hairy, then I'll abort and it can wait for a future PR.

Reply:
I did a quick pass, but I'm gonna push it up as a separate PR for easier changelog tracking.  Also going to revert the spotbugs fix I did, and include it in that versions PR.

Reply:
Sounds good to me!